### PR TITLE
feat(perf): improve preview loading

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
* Adds a limit of parallel preview requests to prevent the server from being overrun with requests
* Cancels ongoing and queued preview requests when they are aborted (e.g. when leaving the page while previews are loading)
* Adds lazy loading for space images
* Prevents the image resource from being fetched for each space
* Uses caching for space images

Also introduces a global `loadPreview` method and removes the visibility observers in each view. Previews are now loaded event-based instead, which should result in a better performance overall.

## Related Issue

- refs https://github.com/owncloud/ocis/issues/10054

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-app-files/src/components/Search/List.vue [hunk 1]
```diff
@@ -108,7 +108,13 @@
           :fields-displayed="['name', 'size', 'tags', 'mdate']"
           :resource-dom-selector="resourceDomSelector"
           @file-click="triggerDefaultAction"
-          @row-mounted="rowMounted"
+          @item-visible="
+            loadPreview({
+              space: getMatchingSpace($event),
+              resource: $event,
+              ignoreAlmostEmptyTxtFiles: true
```
Comment (kulmann) [inline packages/web-app-files/src/components/Search/List.vue:115]: Huh, does this feature still exist? I thought we killed it.

Comment (JammingBen) [reply packages/web-app-files/src/components/Search/List.vue:115]: We ignore almost empty text files in the file lists apparently. Should I kill it? 😂

Comment (kulmann) [reply packages/web-app-files/src/components/Search/List.vue:115]: Yeah I remember when we built the feature. But the added value is minimal while adding unnecessary complexity. So I thought we killed it already. If not I'd vote for removing the feature from the code base.

File: packages/web-pkg/src/components/SideBar/Spaces/Details/SpaceDetails.vue [hunk 2]
```diff
@@ -167,16 +142,36 @@ export default defineComponent({
...
+          return
+        }
+
+        const imageResource = buildSpaceImageResource(unref(resource))
+        spaceImage.value = await loadPreview({
+          space: unref(resource),
+          resource: imageResource,
+          dimensions: ImageDimension.Tile,
+          processor: ProcessorType.enum.fit,
+          cancelRunning: true,
```
Comment (kulmann) [inline packages/web-pkg/src/components/SideBar/Spaces/Details/SpaceDetails.vue:158]: Really an edge case, but couldn't it happen now that you show space details from the space overview while previews are still loading and then that would kill all the remaining preview requests in the space overview?

Comment (JammingBen) [reply packages/web-pkg/src/components/SideBar/Spaces/Details/SpaceDetails.vue:158]: `cancelRunning` would only cancel preview requests within this component. The space overview, while also using the `useLoadPreview` composable, has a different "instance" of it. So technically there are 2 separate tasks.

Comment (kulmann) [reply packages/web-pkg/src/components/SideBar/Spaces/Details/SpaceDetails.vue:158]: Aha! nice, thanks for the explanation 👍

**Review by kulmann — APPROVED** [review]
So awesome 😍

**Pull request merged**